src/runtime/hexagon_dma_context.h

```diff
@@ -0,0 +1,129 @@
+    t_work_buffer* pwork_buffer;                // Work Buffer
+} t_resource_per_frame;
+
+/**
+ *  This Below Structure is at DMA Resource Granularity
+ *  Capture all the Detials at the DMA resource Level */
+typedef struct {
+    bool in_use;                                // To identify if the DMA Resource is in use or free
+    int resource_id;                            // ID for the DMA Resource to Distinguinsh from other resources
+    t_halide_dma_session session;               // PSession Holds both the DMA Read/Write handles or Either of them
+    bool dma_write;                             // flag to indicate DMA write is also allocated
+    t_resource_per_frame *pframe;               // We donot allocate memory for this,just associate the current frame this Engine handles
+} t_dma_resource;
+
+/**
+ *  Capture Detials of available DMA  Resources */
+typedef struct {
+    bool in_use;                                // To Identify if the H/W thread is in use or Free
+    int dma_set_id;                             // To Distinguinsh various sets of DMA engines
+    int ndma_read_engines;                      // Number of DMA Enginers in use for Read in this SET
+    int ndma_write_engines;                     // Number of DMA Enginers in use for Write in this SET
+    t_dma_resource *pdma_read_resource;         // Allocate this based on the number of Read DMA Engines i.e nDmaReadEngines. Currently 1
+    t_dma_resource *pdma_write_resource;        // Allocate this based on the number of Write DMA Engines i.e nDmaWriteEngines. Curently 1
+} t_set_dma_engines;
+
+/** Frame Table
+ * For Fast Seraching of the Frame in the DMA structures */
+typedef struct {
+    uintptr_t frame_addr;                       // Virtual Address of the Frame
+    int dma_set_id;                             // ID for the DMA Set used for this Frame 
+    int dma_engine_id;                          // ID of DMA engine used for this Frame
+    int frame_index;                            // The Frame ID used to disntinguish various Frames
+    bool read;                                  // To distinguish if the Frame is For Read or write
+    int work_buffer_id;                         // The Work Buffer attached to this Frame 
+    bool update;                                // to indicate if update is in progress
+} t_frame_table;
+
+/**
+ * Hexagon DMA transfer user_context */
+struct t_dma_context {
+public:
+    int nframes;                                       // Total Number of Frames, used to maintain frame Table 
+    int current_frame_index;                           // Current Frame Index
+    int frame_cnt;                                     // A Global count for indexing the Frames 
+    int fold_cnt;                                      // A Global Count for indexing the Fold 
+    int num_threads;                                   // Number of Software Threads = No of Tiles
```

Comment:
Two comments on this (num_threads):

- First, I highly doubt we want this here. Parallelism should be handled in the Halide schedule, the runtime should not have to worry about it.

- Second, this is the only reference in the PR to 'num_threads', other than the one function that trivially assigns it. It's impossible to tell if this is something reasonable or not when it's basically an unused declaration. Can you remove any unused stuff like this?